test/credentials.spec.js

```diff
@@ -1474,6 +1490,48 @@
         });
       });
 
+      it('supports chaining in absense of web_identity_token_file and has source_profile ', function(done) {
+        delete process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
+        helpers.spyOn(AWS.util, 'getProfilesFromSharedConfig').andReturn(
+          {
+            'source': {
+              'web_identity_token_file': 'cfgTokenFileSource',
+              'role_arn': 'cfgRoleArnSource'
+            },
+            'default': {
+              'source_profile': 'source',
+              'role_arn': 'cfgRoleArnDefault'
+            }
+          }
+        );
+        var credentials = new AWS.TokenFileWebIdentityCredentials();
+        credentials.refresh(function() {
+          expect(fs.readFileSync.calls[0]['arguments'][0]).to.equal('cfgTokenFileSource');
+          const sourceCredentials = {
+            AccessKeyId: 'AccessKeyIdSource',
+            SecretAccessKey: 'SecretAccessKeySource'
+          };
+          var assumeRoleSpy1 = helpers.spyOn(credentials.service, 'assumeRoleWithWebIdentity').andCallFake(function(params, cb) {
+            return cb(null, {
+              Credentials: sourceCredentials
+            });
+          });
+          var assumeRoleSpy2 = helpers.spyOn(credentials.service, 'assumeRole').andCallFake(function(params, cb) {
```

Comment:
nitpick: I'm not a big fan of the naming here. How about `assymeRoleSpy` and `assumeRoleWithWebIdentitySpy`?

Reply:
Renamed in https://github.com/aws/aws-sdk-js/pull/2837/commits/5b24b41216f8d2ee939d8b81e8a20cf21a2b4432